app/scripts/controllers/transactions/index.js

```diff
@@ -376,6 +376,7 @@ class TransactionController extends EventEmitter {
       // add nonce to txParams
       // if txMeta has lastGasPrice then it is a retry at same nonce with higher
       // gas price transaction and their for the nonce should not be calculated
+      log.debug('customNonceValue', txMeta.customNonceValue)
```

Comment:
I'd assume this property exists at this point, but I don't see this log anywhere? not sure what makes the most sense to debug this?

Reply:
Were you looking in the background console for the log? To see it you need to:
- go to `chrome://extensions/` or the Brave equivalent
- turn "Developer Mode" on
- under the MetaMask app you are building, inspect the "Background page"
- click the "console" tab

Reply:
Okay, I got things _sorta_ working... was able to kick off a transaction with a custom nonce of `4`:

![image](https://user-images.githubusercontent.com/675259/64303456-3d0e5f80-cf56-11e9-8775-77bb7dd5785e.png)

however it's just been pending this whole time... I'm not sure what the requirements of `nonce` are... trying strings like 'testing123' (as an example) just seems to break things (transaction doesn't go through and I get a bunch of errors).

Reply:
In fact, I think what you have is working. That the tx is in pending the whole time is due to how transactions are handled by the ethereum blockchain. If the network only has record of nonces 1 and 2 for your account, and you custom edit a new transaction to nonce 5, that new one will remain pending until transactions with nonces 3 and 4 are successfully confirmed.

Read this for a helpful explanation: https://www.reddit.com/r/ethereum/comments/6ihw6p/can_someone_please_explain_nonce_to_me/dj6r1lk/

You are right, however, that a nonce cannot just be any arbitrary string. It must be a natural number >= 0. So we should prevent other values from being entered in this input field.

Reply:
- [ ] Ensure inputs can only be submitted if nonce is a whole number >= 0, or ensure those are the only input possible in the nonce field